### DISCUSSION TITLE
Dynamic Power limiter regelt zu weit herunter

### DISCUSSION THREAD

**Discussion by Dietemann** [hoylabs/OpenDTU-OnBattery#627] [Q&A]
Hallo an die Experten,
ich betreibe einen HM 1500 OHNE Akku mit vier angeschlossenen Modulen, zwei in SO und zwei in NW Richtung ausgerichtet. Vorher habe ich die Anlage über 1 Jahr mit der Open DTU gesteuert, bis ich im Dezember testweise auf die Open DTU on Battery gewechselt habe.
Eingestellt ist die DTU wie im "Handbuch" beschrieben. Die Erlaubte Einspeisung ist auf -25 Watt gestellt und die Hysterese auf 10 W.
Der Verbrauch wird über einen Hichi Lesekopf an die DTU gefunkt.. dies wird auch genau in der WebApp angezeigt.
Was mir jedoch aufgefallen ist, ist der Fakt, dass die DTU zu weit runterregelt ohne die Solare Leistung auszunutzen, Im Dezember/ Anfang Januar hatte ich schon so eine Vermutung. Heute war schon morgens Früh strahlend Blauer Himmel und minus 2 Grad Celsius. Aus der Erfahrung weiss man schon, was die Anlage liefern müsste. Habe in der WebApp geschaut und der WR war auf 12-13 Prozent geregelt. Es sollten so 220 Watt sein, welches das Haus inkl. der -25 Watt des erlaubten Netzbezuges braucht. 
Die Inverter Gesamtleistung stand so bei 120 Watt, der Netzbezug bei 50-60 Watt. 
Dann habe ich mal den Kaffeevollautomat angeworfen, welcher sich so rund 1200 Watt genehmigt. Die DTU regelte auf die volle 1500 W Leistung des WR hoch, und siehe da- auf einmal stand die Modulleistung auf über 220 Watt- somit wäre meine Grundlast gedeckt.
Vorher hatte ich noch 50-60 Watt aus dem Netz bezogen wo eigentlich noch Potenzial der Module da wäre.
Als die Maschine fertig war, und der WR noch kein neues Limit bekommen hatte speiste ich fast 60! Watt ins Netz ein.
Der Hichi meldete wieder einen geringeren Verbrauch, der WR bekam ein neues Limit 200-210 Watt, und ich bezog wieder meine 50-60 Watt  aus dem Netz, wo Sekunden zuvor noch eingespeist wurde.
Dies habe ich ein Paar mal durchgespielt.
Kurze Frage langer Sinn;)?
Ist hier irgend ein BUG im System, oder liegt es schlichtweg daran dass die Module unterschiedlich ausgerichtet sind und die zwei MPP Tracker aber parallel gleich gedrosselt werden und die DTU nicht erkennt, wieviel Leistung von den einzelnen Trackern kommt?
Die Nord-West Module haben da noch keine große Leistung. Die beiden Tracker werden auf 115-120 Watt gleichermaßen gedrosselt, kommt nur die Leistung von 120 Watt von den Süd-Ost Modulen wird die Differenz  aus dem Netz bezogen, und nicht die WR Leistung angehoben.
Weiss jemand Rat/ Abhilfe, oder sollte man dies an die Jungs weiter geben, die diese Software  auch schreiben?

Danke im Voraus
LG aus Nordhessen

**Comment by Manos1966** [comment]
Ich stelle jetzt hier die Frage, weil ich eine andere Meinung als @spcqike  😉 bezüglich der Aufteilung des Power Limits habe

HM1500,  zwei MPPTs, unterschiedliche Richtungen, keine Batterie.

**Ich** behaupte:

- Das Limit wird von der OpenDTUonBattery an den HM1500 gesendet
- Der HM1500 teilt das Limit durch zwei und sendet dies an jede der zwei MPPTs
- "Die beiden Tracker werden auf 115-120 Watt gleichermaßen gedrosselt, kommt nur die Leistung von 120 Watt von den Süd-Ost Modulen wird die Differenz aus dem Netz bezogen, und nicht die WR Leistung angehoben."

Antwort: "Made by Design" (so ist die Hoymiles konstruiert)

@Dietemann ich habe Kommentare gelesen, dass der Code der OpenDTUonBattery die unterschiedliche Leistung zwischen der zwei MPPT identifizieren kann und entsprechend das Limit anpassen (korrigieren).
Ich habe es fuer nicht moeglich gehalten.
Du hast es bestaetigt.

Es kann aber ein anderer Grund sein:
**Ich** behaupte:

- Die OpenDTUonBattery kann tatsaechlich erkennen ob die MPPTs an Paneele angeschlossen sind oder nicht
- Aber nur wenn sie 0 Watt liefern = keine Solarpanel angeschlossen
- Da aber deine Solarpanels immer ein paar Watt produzieren, werden alle Eingaenge als Belegt identifiziert und die OpenDTUonBattery teilt das Limit gleichmaessig auf die zwei MPTTs

Ich hoffe du hast ein Homematic System, dass die Daten der OpenDTU speichert? 🙈 
Dann konntest du die Werte vergleichen: Stromzaehler Last, HM1500- Limit, AC Leistung, DC Leistung Paneele 1&2, DC Leistung Paneele 3&4

Eine Verstaendniss Frage habe ich aber noch:
"Die DTU regelte auf die volle 1500 W Leistung des WR hoch, und siehe da- **auf einmal stand die Modulleistung auf über 220 Watt**- somit wäre meine Grundlast gedeckt."
Also die HM1500 bekam das Limit 100% (1500W), wieviel hat sie tatsächlich geliefert, nur 220W ?

**Reply by Manos1966** [reply]
#590 Wiki Frage: bei der Berechnung des Power Limits, die OpenDTU prüft wie viele Eingänge produzieren (>2W liefern) und skaliert entsprechend. 


und bevor ich es vergesse, es gibt ein Grund, warum das Projekt heißt OpenDTU**onBattery** - es war nicht die Idee, es mit unterschiedlichen Paneele-Richtungen zu nutzen 😉

**Reply by spcqike** [reply]
Was für eine andere Meinung denn? Ich hätte das gleiche raten wie du. Die SO Paneele hängen an MPPT1, die NW Paneele an MPPT2. Da der hoymiles das Limit intern einfach auf beide tracker zu gleichen Teilen aufteilt, und ein Eingang mehr liefern kann, produziert man in Summe nicht das, was man gern hätte. 

Ich hatte mal einen PR offen, der das erkannt und behoben hätte. Aber der wurde zu seiner Zeit abgelehnt. (Opendtu onBattery ist ja auch mehr für den Batterie betrieb gedacht, wo das Problem eigentlich nicht vorkommt. Und die Änderung sei wohl „zu komplex“ und ggf. inkompatibel mit anderen setups. Ich konnte es nur mit meinem HM-600 testen) 

Wenn die Paneele wirklich so angeschlossen sind, würde es bereits helfen, jeweils 1 SO und 1 NW pro MPPT zu betreiben.

**Reply by Dietemann** [reply]
Es hat mir keine Ruhe gelassen und ich habe mir Gedanken gemacht wie es mit den festgestellten Erkenntnissen trotzdem funktionieren könnte.
Heute habe ich Nachtschicht, und konnte Tagsüber noch etwas testen und meine, und die Behauptung von @Manos1966 bestätigen.
-Die OpenDTUonBattery erkennt, ob an die MPPTs an Paneele angeschlossen und Spannung anliegt, sonst würde ein Fehlercode geschrieben werden.
-Da die angeschlossenen NW Module auch mit diffusem Licht produzieren, bei mir um ca. 9:00 Uhr um die 20 Watt, wird die Leistung dazu addiert.
OpenDTUonBattery teilt das Limit gleichermaßen auf die zwei MPTTs auf, diese "interessiert" es aber nicht, ob am anderen Ende die gewünschte Leistung auch raus kommt.. -Ist tatsächlich dem Konstrukt WR geschuldet, der die Leistung am Mppt regelt und nicht am Ausgang.
Deshalb habe ich mir gedacht, dass ich die Leistung pro Eingang/Mppt künstlich anhebe, damit die DTU den Wechselrichter animiert  eben dort mehr zu produzieren wenn möglich.
So habe ich den Angestrebten Netzbezug bei mir  auf -150 Watt anstelle der -25 Watt gesetzt, um eine Anhebung der Leistung zu erzwingen um mehr ins "Netz" zu pumpen.
Die SO Module sollen mehr liefern, um die geringe Leistung der NW Paneele auszugleichen, da dort nie die errechnete Leistung erreicht wird.
Es schien die Sonne und meine These ging auf.
Die DTU regelte beide MPPT auf ca. 23 % also rund 350 Watt. Mein Grundbedarf liegt so bei 200-210 Watt. Der SO Tracker lieferte etwas über 190 Watt, die NW Module ca. +/-50 Watt. 
Mit dieser Einstellung schubste die Anlage dann noch ca. 14 Watt ins Netz ein. -Für mich ein zufriedenstellendes Ergebniss.
Nichts desto trotz könnte die Weiterentwicklung der DTU noch so perfektioniert werden, dass die DTU auch nachprüft, ob die geforderte Leistung auch am ende raus kommt, und ggf. nachregelt.
Mal schauen, was da noch von den findigen Köpfen noch kommt, die Open DTU wird stätig besser

**Reply by spcqike** [reply]
Wenn es so für dich erstmal läuft, ist ja gut. Dann hast du erstmal ne Lösung 

wenn ich Zeit finde, kann ich ja mal probieren, die Regelung, die ich bereits einmal hatte, wieder in einen fork meinerseits zu packen. Leider habe ich das Repo in der Zwischenzeit gelöscht. Aber so viele Änderungen waren es glaube ich gar nicht. 

Testen kann ich es leider nur mit meinem HM-600, der nur zwei Eingänge und zwei MPPTs hat.

**Comment by Manos1966** [comment]
> Wenn die Paneele wirklich so angeschlossen sind, würde es bereits helfen, jeweils 1 SO und 1 NW pro MPPT zu betreiben.

Das koennte sogar die perfekte Loesung sein! 👍 

@Dietemann kannst du es testen?

Rein Informativ: 
[Shading in solar cells two cells in parallel](https://www.youtube.com/watch?v=tWxg7ZtviZs)

PS> @Dietemann falls du die Aenderungen machts, NICHT waehrend die Paneele viel Sonne haben...

**Reply by Dietemann** [reply]
Daran habe ich auch schon gedacht. Allerdings sind es Module mit unterschiedlichen Spannungen. Vor knapp 1,5 Jahren waren 400 Watt Module "state of the art" und dementsprechend noch teuer. Damit habe ich angefangen. Im Herbst kamen dann noch die Nord-West Module mit je 430 WP. Diese haben jedoch eine deutlich  höhere Leerlaufspannung als die anderen, somit fällt so eine Schaltung weg.

**Reply by Manos1966** [reply]
Mist! 🤦‍♂️ 😢

**Reply by Manos1966** [reply]
...wobei, an deiner Stelle wurde ich pruefen lassen dein HM-1500 die unterschiedlichen Paneele verkraften kann.
Es sind ja nur 30Wp Unterschied.

**Reply by spcqike** [reply]
@Manos1966 kann man denn Module mit unterschiedlich vielen Zellen parallel schalten? Es geht ja weniger um die Leistung, die 30W, als viel mehr um die Spannungen. Und die sind bei 54 Zellen eben 37V Uoc und 31V Umpp, bei 72 Zellen hingegen 52V Uoc und 44V Umpp. 

Auf welche Spannung soll sich der MPPT da festlegen? Da läuft kein Modul am MPP. 

Bzw, geht das elektrisch überhaupt oder fließt der Strom dann „rückwärts“ durch das schwächere Modul?

**Reply by Manos1966** [reply]
@spcqike Die Situation ist nichts anderes als zwei gleiche Solar Paneele, eins davon ist (durch ein Baum) aber teils verschattet 😉

**Reply by Manos1966** [reply]
![Placing DIFFERENT PV modules in Series or Parallel](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/138a4e3a-fe9a-4a8a-a1c9-5edd3bcd1b11)

Dies bedeutet NICHT, dass der Hoymiles genauso funktioniert. **Ich** behaupte dass es nicht so ist (sonst wurde die OpenDTU nicht die einzeln Spannungs- und Strom-Werte pro Panel geben koennen).

**Reply by spcqike** [reply]
@Manos1966 ich finde dazu unterschiedliche Angaben im Netz. bei zu großen Spannungsunterschieden, sollte man es nicht machen. Wenn man parallele Strings aufbaut, und die resultierenden Stringspannungen identisch sind, ist es wohl kein Problem. in dem Victron Beispiel mit 20V und 24V ist es wohl auch verkraftbar. genauso wie in folgendem Beispiel:
![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/50098985/f6e94bbb-ee0a-4cdf-ae8b-063855c13d86)

bei 30V und 40V (also 54Zellen und 72Zellen) wäre ich mir da aber nicht so sicher. immerhin liefern die 72-Zellmodule einfach mal 33% mehr Spannung, als die 54-Zellmodule.

Ein anderer Aspekt, den man in dem Bild hier aber deutlicher sieht, sind die generellen Verluste. Man kastriert seinen MPPT auf die schwächste Paneelspannung und die Summe der Ströme.

Angenommen alte 400W 54 Zellmodule und neuere 430W 72Zellmodule

![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/50098985/105f4944-06d5-4a75-afe0-1e39578b28a1) ![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/50098985/92819332-08a2-42fc-8d1b-398e27500c66)

die 72 Zellmodule würden also nur mit 31V laufen und dabei nicht mal ihren Nennstrom schaffen (der MPPT liegt ja eben bei 41V)
![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/50098985/49ed1916-1aca-4955-b2f5-ed27ba359dda)

bei ~30V liegen sie also irgendwo zwischen 300W und 350W. (bei unseren Breitengraden und 800W/m² EInstrahlung sogar nur noch bei ~260W.)

also hat man am Ende ein altes 400W Modul und ein neues, auf ~330W "kastriertes" Modul. 

kurz um: ich würde Module mit so unterschiedlichen Spannungen nicht parallel schalten. im Beispilbild oben verliert man durch 2V Unterschied ja bereits 22% der installierten Leistung. in meiner Beispielrechnung mit den 2 Modulen sind es immerhin ~12% (100W von theoretischen 830W)


Nachtrag1:

> Die Situation ist nichts anderes als zwei gleiche Solar Paneele, eins davon ist (durch ein Baum) aber teils verschattet

doch ist es. denn die Spannung bleibt bei der Verschattung nahezu gleich. der Strom bricht ein. (also im Parallelbetrieb). Die Spannung der Module steigt sehr schnell, auch bei Bewölkung und früh am Morgen. Daher ist auch ein Parallelbetrieb von Süd und Nord Strings ohne größere Einbußen möglich. Solange die Spannungen identisch sind. Dann liefert der Südstring eben seine 30A, der Nordstring nur 0.5A, aber beide laufen auf Nennspannung.

Nachtrag2:
in deinem Victron Beispielbild verliert man auch knapp 10% der gesamten Nennleistung. ~22W beim 130W Modul

**Reply by Manos1966** [reply]
@spcqike 
Ziel: der HM1500 muss die Grundleistung korrekt abdecken: 220W
Er hat 400x2+430x2= 1660Wp oder 800-860Wp pro Seite.

Die Verluste sind nicht wichtig, wenn das Ziel erreicht werden kann (Ergebnissorientierung).
Die offene Fragen die ich habe sind nicht die Verluste, sondern wie verhalten sich die Dioden der Paneele, denn (sofern ich verstanden habe)  die "Verluste" die du berechnet hast, sind Energie die irgendwo auf ein Panel absorbiert wird.
Aber wie gesagt, ich habe wenig Ahnung...

Ausserdem, ich gehe davon aus, der HM1500 betrachtet die Eingänge einzeln (die OpenDTU berichtet über jedes Panel einzeln), und nutzt ein MPTT fuer jede zwei.
Das bedeutet (glaube ich) hier gibt es weder Serie, noch Parallel Diskussion.
Also, nur die MPPT Optimierung findet nicht statt! Bei einer theoretische 400+430=830Wp pro Seite und ein Ziel von 220W, ich behaupte es sollte erreichbar sein, auch ohne Optimierung ;)

Das einzige Problem: Ich habe nirgendwo eine Diskussion gefunden, die beschreibt wie der Hoymiles unterschiedliche Paneele auf 1&2 (oder 3&4) handeln kann.

**Reply by spcqike** [reply]
> Ausserdem, ich gehe davon aus, der HM1500 betrachtet die Eingänge einzeln (die OpenDTU berichtet über jedes Panel einzeln), und nutzt ein MPTT fuer jede zwei.

wie sieht denn die Modulspannung der Eingänge aus? Haben Eingang1 und Eingang2 immer die gleiche Spannung? Theoretisch müsste das ja so sein.

> Das bedeutet (glaube ich) hier gibt es weder Serie, noch Parallel Diskussion.

Korrekt. Eingang1 und Eingang2 sind automatisch parallel. Sonst hätten sie ja einzelne MPPTs.

> Also, nur die MPPT Optimierung findet nicht statt!

Doch, die wird stattfinden. 

Ich finde leider kein passendes Beispielbild dazu, aber so in etwa:
![sustainability-13-00830-g016-550](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/50098985/0031de1a-272a-49d7-9d9c-ca92f8323c54)

Die einzelnen MPP-Diagramme würden sich ja überlagern. bis ~ 31V steigt der Strom und damit die Leistung beider Module. ab 31V fällt der Strom des schwächeren Moduls wieder, der Strom des stärkeren Moduls würde weiter steigen bis zu seinem Maximum bei ~41V. irgendwo dazwischen liegt der MPP (Maximum Power Point) des Verbunds. Der MPPT (der Tracker!) versucht diesen zu finden und "stellt seinen internen Widerstand" entsprechend, dass die maximale Leistung erreicht wird. Da der Abfall nach dem MPP aber so rapide ist, und das schwächere Modul sehr schnell nahe 0W produziert, wird die maximale Leistung nicht weit über 31V erreicht werden. Damit begrenzt man eben das stärkere Modul. 

![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/50098985/aa8c1ee9-9bef-40b4-ad9e-ec9ba9619e09) ![300733709-49ed1916-1aca-4955-b2f5-ed27ba359dda](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/50098985/552a4780-0163-4d4b-abb6-1fbb83f6da08)

> Bei einer theoretische 400+430=830Wp pro Seite und ein Ziel von 220W, ich behaupte es sollte erreichbar sein, auch ohne Optimierung ;)

Mag sein, dass einem das genügt. wenn man nur 220W produzieren will. Dennoch begrenzt man seine Paneele nicht unwesentlich. von den theoretischen 830W kann man, in dieser Konstallation, ja maximal 730Wp abrufen (STC Leistung). Unter Realbedingungen (NOCT) verbleiben davon noch gute ~300W und ~260W, also 560W, unter steigender Temperatur (also nicht im klimatisierten Laboraufbau, 20°C mit 1m/s Wind) weiter fallend. 
Ja, selbst 560W Nennleistung (nach der AC Wandlung, unter Annahme von 95% Effizienz also 530W), reichen, für 220W. 

Dennoch würde ich das wahrscheinlich so nicht machen :) die bis hier her theoretischen Verluste von 100W hat man ja auf jedem Tracker. also 200W in Summe. Das ist bereits ein halbes Paneel "Verlust" (Leistung, die aufgrund des Mixes einfach nicht abgerufen werden kann). Wäre mir persönlich einfach zu schade 😄 und die Wolken und das schlechte Wetter gibt es ja auch noch. das stärkere Paneel wird immer Einbußen hinnehmen müssen und man hat immer weniger, als man hätte haben können. 

> Das einzige Problem: Ich habe nirgendwo eine Diskussion gefunden, die beschreibt wie der Hoymiles unterschiedliche Paneele auf 1&2 (oder 3&4) handeln kann.

ich kann es nicht testen oder bestätigen, aber in der Theorie beschreibt mein Text hier das, was passieren sollte.

**Reply by Manos1966** [reply]
@spcqike Hmmm.... kannst du es auch auf Deutsch erklären? 🙈 🤣 🤣 🤣 🤣 🤣 🤣 🤣 

Das war etwas "heavy" fuer mich 🤣 
Es ist ein sehr komplexes Thema und ich habe (noch) keine Zeit dafuer gewidmet - vor allem weil ich es einfach umgehen kann wenn ich gleiche PV-Module installiere (ich werde den Teufel tun unterschiedliche Module zusammenzuschließen 🤣). 
Und meine Hauptsorge ist erst die Hot-Spots Gefahr zu verstehen, bevor ich mich mit dem Thema "Effizienz gemischte PV-Module" beschaeftige. 
![Capture](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/4494909e-57c2-4ada-8519-11ee96d4993f)

Du hast Recht, 1&2 sowie 3&4 haben jeweils die gleiche Spannung
![Untitled](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/dc55030f-7408-45fc-be90-e855cccb9d75)


Nachtrag: Nicht mit unserem Fall hier zu tun, aber vielleicht findest du das interessant
Bypass VS Blocking Diodes in a Solar Power System
https://www.youtube.com/watch?v=-x_6OLdAods

**Comment by oldman4U** [comment]
Abend. Ich habe an einem HM-800 zwei Batterie und die Nulleinspeisung mittels DTSU666 und DTU PRO geregelt. Bis zu einem DTU PRO Firmwareupdate letzten Herbst regelte beim Ausfall einer Batterie der zweite Eingang nicht nach und produzierte brav weiter die ihm zugedachten 50%. Seit dem Update verdoppelt der verbliebene Eingang seine Leistung. Hoffe das hilft.
